src/ansys/units/unit.py

```diff
@@ -62,12 +159,8 @@ def __str__(self):
 
     def __mul__(self, __value):
         if isinstance(__value, Unit):
-            temp_dimensions = [
-                dim + __value.dimensions[idx] for idx, dim in enumerate(self.dimensions)
-            ]
-            new_dimensions = ansunits.Dimensions(dimensions=temp_dimensions)
-            new_units = new_dimensions.units
-            return Unit(units=new_units)
+            new_dimensions = self.dimensions + __value.dimensions
```

Comment:
We should apply a slight redesign here to nail the concept properly. Multiply here should multiply the dimensions. But _inside_ the `Dimensions` operator implementations, we do the appropriate arithmentic, which in this case means adding values across the two `dict`s.